include/freetds/odbc.h

```diff
@@ -529,10 +529,7 @@ enum {
  * connectparams.h
  */
 
-typedef struct {
-	const char *p;
-	size_t len;
-} TDS_PARSED_PARAM;
+typedef DSTR TDS_PARSED_PARAM;
```

Comment:
I think it would be easier to pass `char*` instead of `const char*` and use the passed string (changing it), all callers uses copies of the original connection strings.